state/cleanup.go

```diff
@@ -1038,13 +1039,13 @@ func (st *State) cleanupForceDestroyedUnit(ctx context.Context, store objectstor
 		}
 		removed, opErrs, err := subUnit.DestroyWithForce(store, true, maxWait)
 		if err == nil {
-			err = applicationService.DestroyUnit(ctx, subName)
+			err = applicationService.DestroyUnit(ctx, coreunit.Name(subName))
```

Comment:
**todo:** If it is safe to not une NewName here (and validate), add a comment that state it.